Add example to Semantic Layer Entity Docs (#7711)
fishtown-analytics/docs.getdbt.com: website/docs/docs/build/entities.md

tags: [Metrics, Semantic Layer]
---

> Entities are real-world concepts in a business such as customers, transactions, and ad campaigns. We often focus our analyses around specific entities, such as customer churn or annual recurring revenue modeling. In our semantic layer models, these entities serve as joins key across semantic models.

Reviewer: ```suggestion
Entities are real-world concepts in a business, such as customers, transactions, and ad campaigns. We often focus our analyses on specific entities, such as customer churn or annual recurring revenue modeling. In our Semantic Layer models, these entities serve as a join key across semantic models.
```

    type: foreign # This can be any entity type key. 
    expr: date_key || '|' || brand_code # Defines the expression for linking fields to form the surrogate key.
```

## Examples

> As mentioned, entites serve as our join keys, using the unique entity name. Therefore, we can join a single `unique` key to multiple `foreign` keys.

Reviewer: ```suggestion
As mentioned, entities serve as our join keys, using the unique entity name. Therefore, we can join a single `unique` key to multiple `foreign` keys.
```

> Consider `date_categories` table with the following columns:

Reviewer: ```suggestion
Consider a `date_categories` table with the following columns:
```

```sql
date_id (primary key)
date_day (unique key)
fiscal_year_name
```

> And a `orders` table with the following columns:

Reviewer: ```suggestion
And an `orders` table with the following columns:
```

```sql
order_id (primary key)
ordered_at
delivered_at
order_total
```

> How might we define our Semantic Layer yml, so we can query `order_total` by `ordered_at` `fiscal_year_name` and `delivered_at` `fiscal_year_name`?

Reviewer: ```suggestion
How might we define our Semantic Layer YAML so that we can query `order_total` by `ordered_at` `fiscal_year_name`, and `delivered_at` `fiscal_year_name`?
```

First, we need to define two `unique` entities in the `date_categories` with the expression set to `date_day`:

```yaml
semantic_models:
- name: date_categories
  description: A date dimension table providing fiscal time attributes for analysis.
  model: ref('date_categories')
  entities:
  - name: date_id
    type: primary

  - name: ordered_at_entity
    type: unique
    expr: date_day

  - name: delivered_at_entity
    type: unique
    expr: date_day

  dimensions:
  - name: date_day
    type: time
    type_params:
      time_granularity: day

  - name: fiscal_year_name
    description: Formatted fiscal year string (e.g. 'FY2025')
    type: categorical
```

> Then, we need to add these same entities as `foreign` keys to our `orders` model and with the expression set to `ordered_at` and `delivered_at`:

Reviewer: ```suggestion
Then, we need to add these same entities as `foreign` keys to our `orders` model, with the expression set to `ordered_at` and `delivered_at`:
```

```yaml
semantic_models:
  - name: orders
    defaults:
      agg_time_dimension: ordered_at
    description: |
      Order fact table. This table is at the order grain with one row per order.
    model: ref('orders')
    entities:
      - name: order_id
        type: primary

      - name: ordered_at_entity
        type: foreign
        expr: ordered_at

      - name: delivered_at_entity
        type: foreign
        expr: delivered_at

    dimensions:
      - name: ordered_at
        expr: ordered_at
        type: time
        type_params:
          time_granularity: day

    measures:
      - name: order_total
        description: Total amount for each order including taxes.
        agg: sum
        create_metric: True
        
 ```

> With this configuration, our semantic models can join on `ordered_at = date_day` via the `ordered_at_entity`, and on `delivered_at = date_day` by the `delivered_at_entity`. To validate our output, we can run:

Reviewer: ```suggestion
With this configuration, our semantic models can join on `ordered_at = date_day` via the `ordered_at_entity`, and on `delivered_at = date_day` via the `delivered_at_entity`. To validate our output, we can run:
```
